### PR TITLE
Fix 22 - Implement test scaffolding

### PR DESCRIPTION
**Pull request by sedge**
This is a work in progress. The main point of contention will be the fixtures and their relationship to our seed scripts.

Remaining work:

* [x] NPM scripts (populate db with test data and run tests at a command)

### CODE REVIEW

**Comment by sedge** [comment]
If this is horribly over-engineered I'll simplify it. I tried to copy `api.wm.o`'s approach as much as possible.

File: test/lib/fixtures/files/get-one.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+var invalidUser = require('../users').invalid;
+
+var validFile = require('./test-files').valid;
+var invalidFile = require('./test-files').invalid;
+
+var validProject = require('../projects').testProjects.valid;
+var invalidProject = require('../projects').testProjects.invalid;
+
+function generateUrl(opts) {
+  return '/user/' + opts.userid + '/projects/' + opts.projectid + '/files/' + opts.fileid;
```
Comment (gideonthomas) [inline test/lib/fixtures/files/get-one.js]: s/`/user`/`/users/`

**Comment by gideonthomas** [comment]
looks good so far!

File: package.json [hunk 2]
```diff
@@ -7,7 +7,8 @@
     "start": "node app",
     "env": "cp env.dist .env",
     "db": "knex migrate:latest && knex seed:run",
-    "test": "lab"
+    "test": "npm run test:setup && lab -t 100 -c --verbose --colors --assert code --timeout 5000",
+    "test:setup": "npm run test:droptables && npm run test:createtables && npm run test:insertdata",
```
Comment (sedge) [inline package.json]: This is incomplete. I was beginning to add this stuff, so don't expect it to work :sob:

**Comment by ashleygwilliams** [comment]
you should be able to use `knex seed:run` to populate the db with test data, is there a reason you aren't doing that?

**Comment by sedge** [comment]
No - I was copying from api. I'll use `knex seed:run` instead

**Comment by ashleygwilliams** [comment]
that may eliminate the need to move the data out of the seeds...

**Comment by sedge** [comment]
Moving the data out of the seeds was intended to eliminate the need to hardcode the tests with values found in the db. I figured it would be a win if they could both reference the same values.

**Comment by ashleygwilliams** [comment]
ah ya, i see your point. :+1:

File: test/api/handlers/files/get-one.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+});
+
+after(function(done) {
+  server.stop(done);
+});
+
+// GET /user/:userid/projects/:projectid/files/:fileid
+experiment('[Get one file]', function() {
+  experiment('[Parameter errors]', function() {
+    test('userid must be valid', function(done) {
```
Comment (ashleygwilliams) [inline test/api/handlers/files/get-one.js]: let's use underscores because that's what they are in the model? user_id

**Comment by gideonthomas** [comment]
r+ merge away

**Pull request merged**